### ISSUE TITLE
Download a file without putting the contents in memory!

### ISSUE THREAD

**Issue by InfinityDevTech** [Steve-xmh/alhc#1]
Hello!

Just a thing, lets say I try to download a file, when I end up doing `request.recv_bytes()` that sticks the contents of the file in ram, IN ITS ENTIRETY.

That could be a huge problem, of which I would like to avoid by potentially converting it to a stream, or adding a function to write it to a file. 

Would this be possible?

**Comment by InfinityDevTech** [comment]
P.S. I love alhc, its been a life saver for file size, and I'm happy to be the first issue!

**Comment by InfinityDevTech** [comment]
I just notiched the download example, but this seems like something that can be merged into the main repository and be made into a single function. Would this be possible?

**Comment by Steve-xmh** [comment]
You can just await a `Request` to turn it into a `Response` that have `AsyncRead` trait implemented, which can let you read binary data as stream:

```rust
let mut response = client.get("https://example.com/hugedata").await?;
let mut buf = [0u8; 1024];
loop {
    let read_bytes = response.read(&mut buf).await?;
    if read_bytes == 0 {
        break;
    }
    // Do something for this chunk of data...
}
```

Current version presented on crates.io doesn't have clear document (my mistake on code architecture) so the `Response` struct doesn't shown, which is what I'm currently working on.

**Comment by InfinityDevTech** [comment]
Ah, that helps a lot! Yeah, there isn't really any documentation on docs.rs, so its impossible to see whats happening without looking at the source code, but asking you would have been easier.

Thank you!

**Comment by Steve-xmh** [comment]
For a new code architecture experiment, I just published a `0.3.0-alpha.1` version to crates.io, should be more clear to use and more stable (probably). Feel free to try it if you have spare time. 😄

**Comment by InfinityDevTech** [comment]
> For a new code architecture experiment, I just published a `0.3.0-alpha.1` version to crates.io, should be more clear to use and more stable (probably). Feel free to try it if you have spare time. 😄

Hey! I tried alpha 3 and its amazing! I also noticed I needed to use futures_util instead of tokio for the `AsyncReadExt`, this is an amazing version, and my binary size is still tiny! Thank you so much!
